pwndbg/gdblib/kernel/kallsyms.py

```diff
@@ -387,15 +398,143 @@ def parse_symbol_table(self):
             self.kallsyms[name[1:]] = (addr, name[0])
 
     def get_token_table(self):
-        tokens = []
-        position = self.token_table
+        if not self.is_uncompressed:
+            tokens = []
+            position = self.token_table
+
+            for num_token in range(256):
+                token = ""
+                while self.kernel_ro_mem[position]:
+                    token += chr(self.kernel_ro_mem[position])
+                    position += 1
+                position += 1
+                tokens.append(token)
+
+        else:
+            tokens = [chr(i) for i in range(256)]
+
+        return tokens
+
+    def find_names_uncompressed(self):
+        # Find the length byte-separated symbol names
+        ksymtab_match = search(
+            rb"(?:[\x05-\x23][TWtbBrRAdD][a-z0-9_.]{4,34}){14}", self.kernel_ro_mem
+        )
+
+        if not ksymtab_match:
+            print(M.error("Failed to find kallsyms"))
+            return None
+
+        kallsyms_names__offset = ksymtab_match.start(0)
+
+        # Count the number of symbol names
+        position = kallsyms_names__offset
+        num_syms = 0
+
+        while position + 1 < len(self.kernel_ro_mem):
+            if (
+                self.kernel_ro_mem[position] < 2
+                or chr(self.kernel_ro_mem[position + 1]).lower() not in "abdrtvwginpcsu-?"
```

Comment:
What is this string `abdrtvwginpcsu`? I know this comes from the vmlinux-to-elf, but do we know what the hell is it? :D

Reply:
It's symbol types in kallsyms. I've added a comment in the code too now.